File: Server/InteractiveEducationalSoftware/InteractiveEducationalSoftware/IES.Models/BusinessModels/StudentLessonProgress.cs
namespace IES.Models.BusinessModels
{
    public class StudentLessonProgress
    {
        public int StudentId { get; set; }
        public int LessonId { get; set; }
        public bool Completed { get; set; }
        public DateTime DateCompleted { get; set; }
    }
}

File: Client/ies/src/app/models/student-lesson-progress.ts
export class StudentLessonProgress{
    studentId: number;
    lessonId: number;
    completed: boolean;
    dateCompleted: Date | null;
    
    constructor() {
        this.studentId = 0;
        this.lessonId = 0;
        this.completed = false;
        this.dateCompleted = new Date();
    }
}